moshi/src/main/java/com/squareup/moshi/KotlinExtensions.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Hide the class from Java consumers.
+@file:JvmName("-KotlinExtensions")
+
+package com.squareup.moshi
+
+inline fun <reified T> Moshi.adapter(): JsonAdapter<T> = adapter(T::class.java)
```

Comment:
What happens if I do 

```kotlin
moshi.adapter<List<String>>()
```

?

My sense is it doesn't work, but may not be avoidable either

Reply:
Mhm, it returns a `CollectionJsonAdapter`, which I assume is a correct one?

Reply:
Yes, but I'm willing to bet it's a raw type and treats it as a `List<Object>`

Reply:
Yep, you're right. But looks like we could overcome this with super type tokens. Not sure how idiomatic it is though. But it's also done like this in [Jackson extensions for Kotlin](https://github.com/FasterXML/jackson-module-kotlin/blob/8ab36da8ba1115d0ef1c2ff20a2f6b1a221fdf73/src/main/kotlin/com/fasterxml/jackson/module/kotlin/Extensions.kt)